src/pages/scale.tsx

```diff
@@ -102,146 +102,137 @@ export function Scale({
             />
           </ButtonGroup>
         </Box>
-        <div
-          style={{
-            flexGrow: 1,
-            display: 'grid',
-            gridTemplateRows: 'auto 1fr',
-            gap: 8
-          }}
-        >
-          <div
-            style={{
-              display: 'flex'
+        <div style={{height: 8}}></div>
+        <ZStack>
+          <Box
+            sx={{
+              display: 'flex',
+              width: '100%',
+              height: '100%',
+              borderRadius: 2
             }}
           >
             {scale.colors.map((_, i) => {
+              const color = getColor(palette.curves, scale, i)
+              const hex = colorToHex(color)
+              const contrast = focusedHex ? getContrast(hex, focusedHex) : undefined
+              const contrastScore = contrast ? getContrastScore(contrast) : undefined
               return (
-                <div
+                <Box
                   key={i}
-                  style={{
+                  tabIndex={0}
+                  onFocus={() => setIndex(String(i))}
+                  sx={{
+                    outline: 'none',
                     width: '100%',
-                    textAlign: 'center',
-                    fontSize: 14
+                    height: '100%',
+                    color: focusedHex,
+                    backgroundColor: hex,
+                    borderTopLeftRadius: i === 0 ? 2 : 0,
+                    borderBottomLeftRadius: i === 0 ? 2 : 0,
+                    borderTopRightRadius: i === scale.colors.length - 1 ? 2 : 0,
+                    borderBottomRightRadius: i === scale.colors.length - 1 ? 2 : 0,
+                    position: 'relative',
+                    fontSize: 1,
+                    display: 'grid',
+                    placeItems: 'end center',
+                    p: 2,
+                    fontWeight: 'bold',
+                    '&::before': {
+                      content: '""',
+                      display: Number(index) === i ? 'block' : 'none',
+                      position: 'absolute',
+                      top: '-8px',
+                      height: Number(index) === i ? 4 : '21px',
+                      left: 0,
+                      right: 0,
+                      backgroundColor: 'var(--color-text)',
+                      borderRadius: 2
+                    },
+                    '&::after': {
```

Comment:
don't really love the semantics here of using a `:before` and `:after` that are both top aligned, but I tried a few different approaches and this seems the most stable and responsive.

Reply:
Eh, I'm fine with it